owncloudApp/src/main/java/com/owncloud/android/MainApp.kt

```diff
@@ -0,0 +1,277 @@
+                    FingerprintManager.getFingerprintManager(activity).onActivityCreated(activity)
+                }
+                // If there's any lock protection, don't show wizard at this point, show it when lock activities
+                // have finished
+                if (activity !is PassCodeActivity &&
+                    activity !is PatternLockActivity &&
+                    activity !is FingerprintActivity
+                ) {
+                    WhatsNewActivity.runIfNeeded(activity)
+                }
+            }
+
+            override fun onActivityStarted(activity: Activity) {
+                Log_OC.d(activity.javaClass.simpleName, "onStart() starting")
+                PassCodeManager.getPassCodeManager().onActivityStarted(activity)
+                PatternManager.getPatternManager().onActivityStarted(activity)
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                    FingerprintManager.getFingerprintManager(activity).onActivityStarted(activity)
+                }
+            }
+
+            override fun onActivityResumed(activity: Activity) {
+                Log_OC.d(activity.javaClass.simpleName, "onResume() starting")
+            }
+
+            override fun onActivityPaused(activity: Activity) {
+                Log_OC.d(activity.javaClass.simpleName, "onPause() ending")
+            }
+
+            override fun onActivityStopped(activity: Activity) {
+                Log_OC.d(activity.javaClass.simpleName, "onStop() ending")
+                PassCodeManager.getPassCodeManager().onActivityStopped(activity)
+                PatternManager.getPatternManager().onActivityStopped(activity)
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                    FingerprintManager.getFingerprintManager(activity).onActivityStopped(activity)
+                }
+                if (activity is PassCodeActivity ||
+                    activity is PatternLockActivity ||
+                    activity is FingerprintActivity
+                ) {
+                    WhatsNewActivity.runIfNeeded(activity)
+                }
+            }
+
+            override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {
+                Log_OC.d(activity.javaClass.simpleName, "onSaveInstanceState(Bundle) starting")
+            }
+
+            override fun onActivityDestroyed(activity: Activity) {
+                Log_OC.d(activity.javaClass.simpleName, "onDestroy() ending")
+            }
+        })
+
+        val newArchModule = module {
+            viewModel { (filePath: String, account: Account, shareTypes: List<ShareType>) ->
+                OCShareViewModel(filePath, account, shareTypes)
+            }
+
+            viewModel { (account: Account) ->
+                OCCapabilityViewModel(account)
+            }
+        }
+
+        startKoin {
+            modules(newArchModule)
```

Comment:
why don't you include androidContext here??

https://insert-koin.io/docs/2.0/documentation/reference/index.html

Reply:
Done